### DISCUSSION TITLE
Attach particles to a rigid body

### DISCUSSION THREAD

**Discussion by BastienRuivo** [NVIDIA-Omniverse/PhysX#61] [Q&A] [answered]
Hi,

I'm trying to make a cloth attached to a rigid body in a scene (like a flag and a pole)
However, I can't find how to attach the cloth to the rigid body (I'm using the PBDClothSnippet)

With some research, I found a particle attach buffer in the ExtGpu namespace that seems to do this, but if it is, how do I add it to my system ?
Thanks !

**Comment by sschirm** [comment, marked as answer]
Hello, 
You can create an attachment in the UI by overlapping the particle cloth and the rigid body or collider, and then Create->Physics->Attachment. This will create an attachment prim under the particle cloth. 
For creating the attachment in script you can refer to the "Deformable Body Attachments" demo script. Creating the attachment on the stage for particle cloth works analogously. Hope that helps!

**Reply by BastienRuivo** [reply]
Hi, thanks for the reply !
If you're talking about the SoftBodyAttachment snippets (I can't find any "Deformable body attachment"), it uses the softBody class which gives me a method to retrives the collision mesh and has a addTetRigidAttachement method if I understood the snippets correctly.
However, In the cloth snippets, cloth are stored inside buffers, and I can't find any similar method with Tetrahedron on the Cloth buffer.

**Reply by sschirm** [reply]
Sorry, I was thinking Omniverse Create...
Roughly, you need to use ExtGpu::PxParticleAttachmentBuffer and call PxParticleAttachmentBuffer::addRigidAttachment for every particle you want to attach (indices matching what you pass to PxParticleBufferDesc), then call PxParticleAttachmentBuffer::copyToDevice.
